#613 - User Request for Private Events

RaccoonBoryvitter opened the pull request:
dev
## GitHub Project

* [Main GitHub Project ticket](https://github.com/EventsExpress/EventsExpress/projects/2)


## Code reviewers

- [ ] @iuriikhrystiuk 

### Second Level Review

- [ ] @sand0 

## Summary of issue

Every private event owner has the possibility to approve or decline a participant. Two buttons to approve or deny users must be implemented in this task, also some styles should be provided for better usability. The approved user appears in the list of visitors and denied user appears in the list of denied users.

## Summary of change

- fixed internal error 500 in API (it was unable to approve or decline user request)
- added styles to buttons in Pending List (provided needed styles)
- deleted redundant Delete Icon from Pending User (this button used to promote users from Pending List, so this point was dangerous)
- fixed missing import in 'approved-list-action.js' (the page had been crashed because of this)

## Testing approach

ToDo

## CHECK LIST
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

RaccoonBoryvitter committed:
#613 - User Request for Private Events

- fixed internal error 500 in API
- added styles to buttons in Pending List
- deleted redundant Delete Icon from Pending User
- fixed missing import in 'approved-list-action.js'

sand0 reviewed (approved):
LGTM

sand0 commented on EventsExpress/ClientApp/src/components/event/pending-users-action.js:
```diff
@@ -1,35 +1,46 @@
+
     return (
         <>
-            {(isMyEvent) &&
```
I guess we shouldn't remove this check

iuriikhrystiuk replied:
We should, because there is no point to promote users to admins from the pending users.

sand0 replied:
This is to verify that the current user has permission to accept request

RaccoonBoryvitter committed:
Handle Event Requests (#613)

- fixed internal error 500 in API (it was unable to approve or decline user request)
- added styles to buttons in Pending List (provided needed styles)
- deleted redundant Delete Icon from Pending User (this button used to promote users from Pending List, so this point was dangerous)
- fixed missing import in 'approved-list-action.js' (the page had been crashed because of this)
